tests/Support/Test/Models/CircularDep.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace RonasIT\Support\Tests\Support\Test\Models;
+
+use Illuminate\Database\Eloquent\Model;
+use RonasIT\Support\Traits\ModelTrait;
+
+class CircularDep extends Model
```

Comment:
let's make more real example, Media model with the Preview relation

```suggestion
class Media extends Model
```